.github/workflows/integration-test.yaml

```diff
@@ -0,0 +1,52 @@
+name: Integration Test
+
+on: [pull_request]
+
+jobs:
+  edx-platform-integration-test:
+    name: Integration with Tutor
+    strategy:
+      matrix:
+        # Open edX Version: Sumac
+        tutor_version: ["<20.0.0"]
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          path: mitx-grading-library
+
+      - name: Adjust permissions to execute Tutor commands
+        run: |
+          chmod 777 . -R
+        shell: bash
+
+      - name: Set Tutor environment variables
+        run: |
+          cat <<EOF >> "$GITHUB_ENV"
+          LMS_HOST=local.edly.io
+          CMS_HOST=studio.local.edly.io
+          TUTOR_ROOT=$(pwd)
+          EOF
+        shell: bash
+
+      - name: Install and prepare Tutor, Codejail and tests
+        run: |
+          pip install "tutor${{ matrix.tutor_version }}"
+          pip install git+https://github.com/edunext/tutor-contrib-codejail
+          tutor config save
+          tutor plugins enable codejail
+          tutor local do init --limit codejail
```

Comment:
Do we need to initialize codejail before launching?

Reply:
If we want to use codejail, we should init it before launch, following these instructions: https://github.com/eduNEXT/tutor-contrib-codejail?tab=readme-ov-file#usage. If not, the codejail container will not find the app armor profile.

About why we need codejail. It is because the MITx course uses the `mitx-grading-library` `python_lib.zip` file to extend the functionality and uses codejail to run the problem in a safe environment. Principally, because without codejail, the unit fails.

![image](https://github.com/user-attachments/assets/58af0ab8-891e-40fe-b5de-0a06d20e22fe)

Reply:
Only to test it, I tried to run it without codejail, and the result was:
![image](https://github.com/user-attachments/assets/d607eecc-c85d-4f4c-bed9-97ff9e30bdf8)

It runs the code using this function: https://github.com/openedx/codejail/blob/master/codejail/safe_exec.py#L240.

The test doesn't break, but as we expect a safe environment to run the Loncapa problems, I will leave the codejail part because it is closer to the real environment.